GVFS/GVFS.Service/ProductUpgradeTimer.cs

```diff
@@ -105,5 +109,24 @@ private bool TryDownloadUpgrade(ProductUpgrader productUpgrader, out string erro
                 }
             }
         }
+
+        private void CleanupDownloadsDirectory(IProductUpgrader productUpgrader)
+        {
+            if (productUpgrader != null)
+            {
+                productUpgrader.CleanupDownloadDirectory();
+                return;
+            }
+
+            try

```

Comment:
It feels weird to delegate to the upgrader to clean up, but to also go whack the directory ourselves if for some reason we couldn't instantiate the upgrader. Why do we do that?

Reply:
Directory cleanup code in ProductUpgradeTimer was not used anywhere. Removed it.